.circleci/config.yml

```diff
@@ -13,27 +13,37 @@ jobs:
       tag: << parameters.java-version >>
     steps:
       - checkout
-      - maven/with_cache:
-          steps:
-            - run: mvn clean test
+      - run: find . -name 'pom.xml' | sort | xargs cat > /tmp/maven_cache_seed
+      - restore_cache:
+          key: maven-{{ checksum "/tmp/maven_cache_seed" }}
+      - run: mvn clean test dependency:go-offline
+      - save_cache:
+          paths:
+            - ~/.m2/repository
+          key: maven-{{ checksum "/tmp/maven_cache_seed" }}
 
   package:
     executor: &default_executor
       name: maven/default
       tag: "11.0"
     steps:
       - checkout
-      - maven/with_cache:
-          steps:
-            - run: mkdir -p ~/artifacts
-            - run: mvn clean package -Dmaven.test.skip=true
-            - run: cp */target/*.jar ~/artifacts
-            - persist_to_workspace:
-                root: ~/
-                paths:
-                  - artifacts
-            - store_artifacts:
-                path: ~/artifacts
+      - run: find . -name 'pom.xml' | sort | xargs cat > /tmp/maven_cache_seed
+      - restore_cache:
+          key: maven-{{ checksum "/tmp/maven_cache_seed" }}
+      - run: mkdir -p ~/artifacts
+      - run: mvn clean package -Dmaven.test.skip=true
+      - run: cp */target/*.jar ~/artifacts
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - artifacts
+      - store_artifacts:
+          path: ~/artifacts
+      - save_cache:
+          paths:
+            - ~/.m2/repository
+          key: maven-{{ checksum "/tmp/maven_cache_seed" }}
```

Comment:
probably worth making all the shared steps in here a command like the orb has but with our custom requirements. We could submit a PR to the orb now that we know what the changes are as well